okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java

```diff
@@ -167,7 +152,25 @@ private static QosIoExceptionAwareOkHttpClient createInternal(
         // dispatcher with static executor service
         client.dispatcher(createDispatcher());
 
-        return new QosIoExceptionAwareOkHttpClient(client.build(), handlerFactory);
+        return new RemotingOkHttpClient(
+                client.build(),
+                () -> new ExponentialBackoff(config.maxNumRetries(), config.backoffSlotSize(), new Random()),
```

Comment:
Could we make this not a supplier? Even if the config live reloads internally, I think it would be nice to not have to pass it around as an arg to `newCallWithMutableState`

Reply:
it has to be a factory since each call needs a fresh, stateful Backoff instance. (this has nothing to do with config reload)

Reply:
Ah OK, I didn't clock the interior mutability in the ExponentialBackoff implementation of `BackoffStrategy`.  Given there's only one implementation right now, what do you think about collapsing that interface?